src/parser.ts

```diff
@@ -1,25 +1,89 @@
+import * as vscode from "vscode";
+
 function normalizeQuery(query: string) {
   return query.replace(/''/g, "'");
 }
 
+function doesFloatParamExistInDeclaration(parameterDeclaration:string, paramName:string){  
+  let paramWithUpperType = paramName + " FLOAT";
+  let paramWithLowerType = paramName + " float";
+  let indexOfUpper = parameterDeclaration.indexOf(paramWithUpperType);
+  let indexOfLower = parameterDeclaration.indexOf(paramWithLowerType);
+
+  if (indexOfUpper != -1 || indexOfLower != -1){
+    return true;
+  }
+
+  return false;
+}
+
+function isParamAlreadyInSingleQuotes(paramValue:string){
+  if (paramValue[0] != "'" && paramValue[paramValue.length] != "'"){
+    return true;
+  }
+
+  return false;
+}
+
+function surroundFloatWithQuotes(parameterDeclaration:string, paramName:string, paramValue:string, value:string, paramDict:Map<string, string>){
+  if (doesFloatParamExistInDeclaration(parameterDeclaration, paramName)){
+    if (isParamAlreadyInSingleQuotes(paramValue)){
+        paramValue = "'" + paramValue + "'";
+        paramDict.set(paramName, paramValue);
+
+        value = paramName + "=" + paramValue;
+        return value;
+    }
+  }
+
+  return value;
+}
+
+function replaceParamInQuery(paramDict:Map<string, string>, normalizedQuery:string){
+  paramDict.forEach((value:string, key:string) => {    
+    normalizedQuery = normalizedQuery.split(key).join(value);
+  })
+
+  return normalizedQuery;
+}
+
 function parseQuery(
   parameterDeclaration: string,
   parameterValue: string,
   query: string
 ) {
   let parameters = parameterDeclaration.split(",");
-  let result = "DECLARE " + parameterDeclaration + "\n";
+  let result = "DECLARE " + parameterDeclaration + "\n";  
+
+  let paramDict = new Map<string, string>();
+  const isSurroundFloatWithSingleQuotesEnabled = vscode.workspace.getConfiguration().get('spexecutesql.surroundFloatWithSingleQuotes');
+  const isInlineParamEnabled = vscode.workspace.getConfiguration().get('spexecutesql.replaceVariablesInQuery');;
 
   parameterValue.split(",").forEach((value, index) => {
     value = value.trim();
 
     if (value[0] !== "@") {
       value = parameters[index].split(" ")[0] + "=" + value;
     }
+
+    let paramName = value.split("=")[0];
```

Comment:
I was thinking instead of passing the dict to the `surroundFloatWithQuotes` you can return the param info and the maintaining dict operation happened only in one place, so will have only one `paramDict.set(paramName, paramValue);`

Reply:
Done